### PR TITLE
[IMP]helpdesk: Added warning for 24 hours format.

### PR DESCRIPTION
**Pull request by kirti-c**
Description of the issue/feature this PR addresses:
Task id - https://www.odoo.com/web#id=36965&view_type=form&model=project.task&action=327&menu_id=4720

Pad - https://pad.odoo.com/p/r.a8472cfacf8a80bfcaf9db3b2df630ba
Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by mba-odoo** [comment]
LGTM

File: addons/resource/models/resource.py [hunk 1]
```diff
@@ -645,6 +645,13 @@ class ResourceCalendarAttendance(models.Model):
     hour_to = fields.Float(string='Work to', required=True)
     calendar_id = fields.Many2one("resource.calendar", string="Resource's Calendar", required=True, ondelete='cascade')
 
+    @api.constrains('hour_to', 'hour_from')
+    def _verify_hours(self):
```
Comment (tde-banana-odoo) [inline addons/resource/models/resource.py]: Please make it wkkr in batch.

Comment (tivisse) [reply addons/resource/models/resource.py]: Better to use meaningful variable names.

``` python
        for attendance in self:
            if attendance.hour_from > 24 or attendance.hour_to > 24:
                raise ValidationError(_("Please enter hours between 1 to 24"))
            if (attendance.hour_to - attendance.hour_from) <= 0:
                raise ValidationError(_("Please use 24 hours format!"))
```

File: addons/resource/models/resource.py [hunk 2]
```diff
@@ -645,6 +645,14 @@ class ResourceCalendarAttendance(models.Model):
...
+        for attendance in self:
+            if attendance.hour_from > 24 or attendance.hour_to > 24:
+                raise ValidationError(_("Please enter hours between 1 to 24"))
```
Comment (bst-odoo) [inline addons/resource/models/resource.py]: string should be : "Hours must be between 1 and 24."

File: addons/resource/models/resource.py [hunk 3]
```diff
@@ -645,6 +645,14 @@ class ResourceCalendarAttendance(models.Model):
...
+            if (attendance.hour_to - attendance.hour_from) <= 0:
+                raise ValidationError(_("Please use 24 hours format!"))
```
Comment (bst-odoo) [inline addons/resource/models/resource.py]: string should be : "Time must be set in the 24 hours format."

**Comment by mart-e** [comment]
Are you sure we want a blocking message? Why not enforce it with onchange.
Setting the from date after the end date happens all the time when you want to modify lines (step 1: change the start, step 2: change the end) and blocking popup are really painful

Something like 0b2b5726bc6 is more friendly don't you think?

**Comment by bst-odoo** [comment]
@mart-e I'm open to suggestions.
I usually don't like that kind of "magic" onchange because people tend to me confused about it.
A blocking message is a PITA but it is clear about what is expected.

**Comment by mart-e** [comment]
I agree about the magic but here it seems easy and what you expect for a calendar : correcting easy mistakes. Check on google calendar, we do +/- the same ;-)

**Comment by bst-odoo** [comment]
If it requires only a few lines, I'm OK.

**Comment by mart-e** [comment]
You can test it on the latest build (300436) http://runbot.odoo.com/runbot/repo/2?search=master-fixes-mat it was not hard ;-)

**Comment by hmo-odoo** [comment]
fixed at https://github.com/odoo/odoo/commit/20f2de82965ed782f14f4154f750853ae5c297f8

**Pull request closed without merging**